peer/record.go

```diff
@@ -0,0 +1,188 @@
+package peer
+
+import (
+	"errors"
+
+	"github.com/gogo/protobuf/proto"
+	"github.com/libp2p/go-libp2p-core/crypto"
+	pb "github.com/libp2p/go-libp2p-core/peer/pb"
+	"github.com/libp2p/go-libp2p-core/record"
+	ma "github.com/multiformats/go-multiaddr"
+)
+
+// The domain string used for peer records contained in a SignedEnvelope.
+const PeerRecordEnvelopeDomain = "libp2p-peer-record"
+
+// The type hint used to identify peer records in a SignedEnvelope.
+// TODO: register multicodec
```

Comment:
Do we have a top-level tracking issue with all TODOs and PRs for this endeavour? If not, we totally should!